sklearn/ensemble/bagging.py

```diff
@@ -265,7 +265,12 @@ def fit(self, X, y, sample_weight=None):
         else:  # float
             max_samples = int(self.max_samples * X.shape[0])
 
-        if not (0 < max_samples <= X.shape[0]):
+        if not (max_samples <= X.shape[0]):
+            warn("max_samples must be less than n_samples." 
+                 " The algorithm will use max_samples=n_samples")
+            max_samples = n_samples
```

Comment:
if `bootstrap=True` then having this constraint on `max_samples` could be removed. However, when `bootstrap=False`, it does not make any sense and I believe an error should be raised.